pkg/deployment/ceph_client.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dataplaneutil "github.com/openstack-k8s-operators/dataplane-operator/pkg/util"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/storage"
+	ansibleeev1alpha1 "github.com/openstack-k8s-operators/openstack-ansibleee-operator/api/v1alpha1"
+)
+
+// ConfigureCephClient ensures the Ceph client configuration files are on data plane nodes
+func ConfigureCephClient(ctx context.Context, helper *helper.Helper, obj client.Object, sshKeySecret string, inventoryConfigMap string, networkAttachments []string, openStackAnsibleEERunnerImage string, ansibleTags string, extraMounts []storage.VolMounts) error {
+
+	haveCephSecret := false
+	for _, extraMount := range extraMounts {
+		if extraMount.ExtraVolType == "Ceph" {
+			haveCephSecret = true
+			break
+		}
+	}
+	if !haveCephSecret {
+		helper.GetLogger().Info("Skipping execution of Ansible for ConfigureCephClient because extraMounts does not have an extraVolType of Ceph.")
+		return nil
```

Comment:
I want this `return nil` to result in the ansibleEE for `dataplane-deployment-configure-ceph-client*` not running. However, when I pass a `DataPlaneNode` CR without an `extraMount` the flow gets stuck re-queueing. 

```
1.6787992745080173e+09	INFO	ConfigureCephClientReady not yet ready, requeueing	{"controller": 
"openstackdataplanenode", "controllerGroup": "dataplane.openstack.org", "controllerKind": 
"OpenStackDataPlaneNode", "OpenStackDataPlaneNode": {"name":"edpm-compute-
0","namespace":"openstack"}, "namespace": "openstack", "name": "edpm-compute-0", "reconcileID": "7ab3d851-
a3a1-4b23-bfc4-44d09509479e"}
```
When I pass a `DataPlaneNode` CR with an `extraMount` the patch works as I had hoped and moves on to `InstallOpenStack`. 

How can I have two flows based on what's in the CR?

```
if extraMounts:
  RunOS --> ConfigureCephClient --> InstallOpenStack
else:
  RunOS --> InstallOpenStack
```

Reply:
you can skip in `deployment.go` the `conditionalDeploy` (where deployName =  ConfigureCephClient) if extraMounts is empty.